### ISSUE TITLE
Correctly trusting proxies / Documentation of

### ISSUE THREAD

**Issue by zsturgess** [hslavich/OneloginSamlBundle#114]
I've had some issues recently using this bundle with Symfony 4.4 when deploying applications behind a TLS-terminating reverse proxy (such as AWS ELB, Heroku, etc.). I've figured out that I need to tell onelogin-saml to trust the proxy (as it doesn't use the symfony configuration for this).

I fixed my issue by adding the following code in my `public/index.php`:
```
if ($trustedProxies = $_SERVER['TRUSTED_PROXIES'] ?? $_ENV['TRUSTED_PROXIES'] ?? false) {
    Request::setTrustedProxies(explode(',', $trustedProxies), Request::HEADER_X_FORWARDED_ALL ^ Request::HEADER_X_FORWARDED_HOST);
+   \OneLogin\Saml2\Utils::setProxyVars(true); # Tell onelogin-saml to trust the proxy, too
}
```

I have 2 questions as a result:

1. Is there a "better" way to do this? I think my configuration above will make onelogin-saml trust *all* proxies, regardless of which proxies are actually trusted by the symfony configuration. (This is OK for my setup as all access is via the reverse proxy, but this may not be the case every time)
2. Should this be documented somewhere? If so, where? (Perhaps under configuration in README.md?)

**Comment by arturslogins** [comment]
Thanks you, you save me 

I have the same issue if i use revers proxy

**Comment by ragusa87** [comment]
I had the same issue and found the same fix, this should definitely go into the documentation.
The is no better way to do it, the problem is that the underlying library uses vanilla PHP like $_POST, and not Symfony component.

**Comment by lpeabodygenuine** [comment]
Yeah I ran into this as well. Should definitely be a configuration option for the package so it can be overridden per-environment.
